tensilelite/Tensile/KernelWriter.py

```diff
@@ -2068,85 +2071,8 @@ def kernelBody( self, kernel, tensorParametersA, tensorParametersB ):
     fs = self.functionSignature()
     moduleKernelBody.addSignature(fs)
 
-    self.defineVariableSgprs(kernel)
-
-    ####################################
-    # Local Read Addresses
-    ####################################
-    lralwaMod = Module("local")
-
-    # C regs are not used during initialization so mark them as available -
-    # we will claim then just before the start of the unroll loop:
-    self.vgprPool.add(self.states.a.startVgprValu , \
-        self.states.lastValuAB - self.states.a.startVgprValu , "ValuAB") # Add as available
-    lralwaMod.addComment0("init: add vgpr [%u...%u) to pool" % \
-                         (self.states.a.startVgprValu, self.states.lastValuAB+self.states.a.startVgprValu))
-
-    self.vgprPool.add(self.states.c.startVgprValu, \
-      self.states.c.numVgprValu, "ValuC-Block") # Add as available
-    lralwaMod.addComment0("init: add vgpr [%u...%u) to pool" % \
-                         (self.states.c.startVgprValu, self.states.c.startVgprValu+self.states.c.numVgprValu))
-
-    numAccvgprs = self.states.totalAgprs
-    self.agprPool.add(0, numAccvgprs, "ValuC-Block")
-    lralwaMod.addComment0("init: add agpr [%u...%u) to pool" % \
-                         (0, numAccvgprs))
-
-    lralwaMod.addComment2("Local Read Addresses")
-
-    # tile assignments
-    lralwaMod.addComment1("local read addresses: tile assignments a/b")
-    lralwaMod.add(self.lraTileAssignment(kernel, tensorParametersA, tensorParametersB))
-
-    tPM = tensorParametersA["tpsMetadata"] if tensorParametersA["is_sparse"] else tensorParametersB["tpsMetadata"]
```

Comment:
you remove tPM here which will be used in the following code segment:
localReadCodeM, packCodeM = self.localReadDo(kernel, plrIdx*self.states.numIterPerCoalescedReadMetadata, iui*self.states.numReadsIterCoalescedMetadata, espi, tPM)
in line 2148.

Reply:
fixed